pandas/tseries/tools.py

```diff
@@ -483,6 +484,132 @@ def parse_time_string(arg, freq=None, dayfirst=None, yearfirst=None):
+        - If 'coerce', then invalid parsing will be set as None
+        - If 'ignore', then invalid parsing will return the input
+
+    Returns
+    -------
+    datetime.time
+    """
+    from pandas.core.series import Series
+
+    def _convert_listlike(arg, format):
+
+        if isinstance(arg, (list, tuple)):
+            arg = np.array(arg, dtype='O')
+
+        elif getattr(arg, 'ndim', 1) > 1:
+            raise TypeError('arg must be a string, datetime, list, tuple, 1-d array, or Series')
+
+        arg = com._ensure_object(arg)
+
+        if infer_time_format and format is None:
+            format = _guess_time_format_for_array(arg)
+
+        times = []
+        if format is not None:
+            for element in arg:
+                try:
+                    times.append(datetime.strptime(element, format).time())
+                except:
+                    if errors == 'raise' or errors == 'ignore':
+                        raise ValueError("Cannot convert %s to a time with "
+                                         "given format %s" % (element, format))
+                    else:
+                        times.append(None)
+        else:
+            formats = _time_formats[:]
+            format_found = False
+            for element in arg:
+                time_object = None
+                for time_format in formats:
+                    try:
+                        time_object = datetime.strptime(element,
+                                                        time_format).time()
+                        if not format_found:
+                            # Put the found format in front
+                            formats.insert(0, formats.pop(
+                                              formats.index(time_format)))
+                            format_found = True
+                        break
+                    except:
+                        continue
+
+                if time_object is not None:
+                    times.append(time_object)
+                elif errors == 'raise' or errors == 'ignore':
+                    raise ValueError("Cannot convert arg %s to a time.  Pass "
+                                     "in format or add additional formats with "
+                                     "add_time_format" % arg)
+                else:
+                    times.append(None)
+
+        return times
+
+    try:
+        if arg is None:
+            return arg
+        elif isinstance(arg, time):
+            return arg
+        elif isinstance(arg, Series):
+            values = _convert_listlike(arg._values, format)
+            return Series(values, index=arg.index, name=arg.name)
+        elif isinstance(arg, ABCIndexClass):
+            return _convert_listlike(arg, format)
+        elif com.is_list_like(arg):
+            return _convert_listlike(arg, format)
+
+        return _convert_listlike(np.array([arg]), format)[0]
+    except:
```

Comment:
It's to cover the `"ignore"` case for casting errors in which case `arg` is returned.  I did it here as things are cast to a numpy array and so your original argument really isn't returned.  However, for now I can remove that casting and then check for `"ignore"` when the casting fails and return the unmolested `arg` at that point.